torch/_inductor/graph.py

```diff
@@ -1319,27 +1319,42 @@ def debug(msg: str) -> None:
                 n.meta["val"], torch.Tensor
             ):
                 strides = n.meta["val"].stride()
-                dense = torch._prims_common.is_non_overlapping_and_dense(n.meta["val"])
-                unbacked_symbols_in_strides = len(free_unbacked_symbols(strides)) > 0
-                # requiring a stride order for a non-dense output wouldn't
-                # recreate the same strides, and would fail with view, defer for now.
-                if not unbacked_symbols_in_strides and dense and len(strides):
-                    stride_order = ir.get_stride_order(strides)
-                    if (
-                        len(result.get_size()) == 4
-                        and n in self.nodes_prefer_channels_last
-                        and n.name not in self.user_visible_outputs
-                        and not is_input_for_as_strided
-                    ):
-                        stride_order = ir.NHWC_STRIDE_ORDER
 
+                if len(strides):
                     allow_padding = (
                         n.name not in self.user_visible_outputs
                         and not is_input_for_as_strided
                     )
-                    result = ir.ExternKernel.require_stride_order(
-                        result, stride_order, allow_padding=allow_padding
-                    )
+                    if not isinstance(result.data, ir.ReinterpretView):
+                        result = ir.ExternKernel.require_exact_strides(
+                            result, strides, allow_padding=allow_padding
```

Comment:
if we move this after strides' update, unexpected success for `test_torchinductor_codegen_dynamic_shapes.py -k test_conv2d_channels_last_dynamic_shapes`. An additional conv part added to the existing cpu kernel.